src/cms/views/events/event_list_view.py

```diff
@@ -94,6 +95,8 @@ def get(self, request, *args, **kwargs):
         # all events of the current region in the current language
         events = region.events.filter(archived=self.archived)
```

Comment:
This method needs some refactoring. But if you like you could move this line right in front of the paginator and only set it if it is not set by any of the many if branches.

Reply:
@svenseeberg I'm not sure if that works, since the `events` variable is not completely replaced in the if-branches, but only filtered further.
However, in the initial solution @david-venhoff implemented it similar like you suggested (see [this comment](https://github.com/Integreat/integreat-cms/pull/929#discussion_r709652412)), but I requested a change earlier.

In view of our performance problems, I even have a completely new suggestion: instead of implementing the search as a model method, we could define the search function as a QuerySet method which can be chained to other database filters and thus reducing the amount of queries used:
```
class EventQuerySet(models.QuerySet):
    def search(self, language_slug, query):
        return self.filter(
            translations__language__slug=language_slug,
            translations__title__icontains=query,
        )

class Event(models.Model):
    ...
    objects = models.Manager.from_queryset(EventQuerySet)()
    ...
```
Then, it could be used e.g. like this:
```
events = region.events.filter(archived=self.archived)
...
query = event_filter_form.cleaned_data["query"]
if query:
    events = events.search(language_slug, query)
```
and in the end, only one sql would be performed, since [querysets are lazy](https://docs.djangoproject.com/en/3.2/topics/db/queries/#querysets-are-lazy)... what do you think of this approach?
Since all models have slightly different search filters, we would probably need one custom QuerySet class for each model...

Reply:
Oh sorry, I did not read carefully enough the following lines. Just merely saw the `events =`

Reply:
Should this be implemented for this pr, or should we open an issue for that?

Reply:
That makes sense. However, I am not sure how to handle the ajax search with this implementation. One option would be to keep it as it is right now, but that would leave us with some duplicate search methods (one version to search the objects, and one version to search the object translations). Or we could remove the old search methods and only use the new `QuerySet` based methods, but then there would be two sql queries required for the ajax search.

Reply:
Ah okay, sorry, I wasn't aware that this requires such substantial changes. In this case, I think we should just remember this idea for later (maybe opening a new refactoring issue with low priority) and merge the current solution. The user experience of the search is already pretty good and a huge step towards the migration :+1: